### PR TITLE
Clean 33/Dispenser API requests part 2

### PR DESCRIPTION
**Pull request by Mikmik28**
## Tickets

https://app.plane.so/mashupgarage/projects/6cabd2bf-26c5-4105-846f-72014688799f/issues/7180dcdc-889c-48b2-9a82-bd7154296707
 
## Changes

- [x] Add Menu API request
- [x] Add Dispenser API requests
- [x] Add Machine Actions API requests

| HTTP method | APIs | Params/Body |
|---|---|---|
| Machine Actions |---|---|
| POST | `startDrinkDispensing` | `dispenser` type: string (dispenser name); `size` type: string (SIZE_OPTIONS: "Small", "Large") |
| POST | `stopDrinkDispensing` | `dispenser` type: string (dispenser name) |
| POST | `cleanDispenser` | `dispenser_name` type: string; `mode` type: number (The mode of the cleaner: 0 (off), 1, 2, or 3) |
| POST | `setHeater` | `dispenser_name` type: string; `heater_status` type: boolean (The status of the heater: False (off), True (on)) |
| POST | `setTempRegulation` | `dispenser_name` type: string; `temperature_regulation` type: boolean (Whether to do temperature regulation: False (off), True (on)) |
| POST | `turnOnTap` | `dispenser_name` type: string |
| POST | `turnOffTap` | `dispenser_name` type: string |
|---|---|---|
| Dispenser |---|---|
| GET | `checkCupPresence` | `dispenser` type: string |
| GET | `getTemperature` | `dispenser` type: string |
| GET | `getThermosWeight` | `dispenser` type: string |
| PUT | `setHeaterStrength` | `dispenser` type: string; `strength` type: number (The strength of the heater: 0 (off), 1 (25% ON), 2 (50% ON), 3 (100% ON)) |
| PUT | `setPump` | `dispenser` type: string; `duration` type: number (duration = 0: disable, stop immediately; duration: 1 ~ 20 - Enable for 1 ~ 20 seconds) |
| PUT | `setCleaner` | `dispenser` type: name; `mode` type: number (The mode of the cleaner: 0 (off), 1, 2, or 3)

## Screenshots (In-progress)
| APIs | Data |
|---|---|
| `getMenuItems` | image |
| `startDrinkDispensing` | image |
| `stopDrinkDispensing` | image |
| `cleanDispenser` | image |
| `setHeater` | image |
| `setTempRegulation` | image |
| `turnOnTap` | image |
| `turnOffTap` | image |
| `dispenserTest` | image |
| `checkCupPresence` | image |
| `getTemperature` | image |
| `getThermosWeight` | image |
| `getVersion` | image |
| `setHeaterStrength` | image |
| `setPump` | image |
| `setCleaner` | image |

### CODE REVIEW

File: src/api/dispenser.ts [hunk 1]
```diff
@@ -1,22 +1,146 @@
...
+
+// Menu
+export const getMenuItems = async () => {
+  const endpoint = `${baseUrl}/api/dispenser/menu-items`
+
+  return axios.get(endpoint).then(res => res.data).catch(err => err.response.data.error)
+}
+
+// Machine Actions
+export const startDrinkDispensing = async (dispenserName: string, size: string) => {
```
Comment (PaoloLlenado) [inline src/api/dispenser.ts:14]: @Mikmik28 the `dispenserName` here is for Tap-A and Tap-B right? 

If it's only going to be one of those 2 options, maybe we can make the type into `"Tap-A" | "Tap-B"`

File: src/api/dispenser.ts [hunk 2]
```diff
@@ -1,22 +1,146 @@
...
+// }
+
+export const setPump = async (dispenserName: string, duration: number) => {
+  const endpoint = `${baseUrl}/api/dispenser/set-pump`
+  const body = JSON.stringify({ dispenser: dispenserName, duration: duration })
+
+  return axios.put(endpoint, body).then(res => res.data).catch(err => err.response.data.error)
+} 
+
+export const setCleaner = async (dispenserName: string, mode: number) => {
```
Comment (PaoloLlenado) [inline src/api/dispenser.ts:121]: Can we put this closer to the other `clean_dispenser` function and add comments to tell the difference between the 2 functions?

Comment (Mikmik28) [reply src/api/dispenser.ts:121]: ayt ayt 👍

File: src/api/dispenser.ts (hunk 1 above)
Comment (Mikmik28) [reply src/api/dispenser.ts:14]: Note: as discussed on the call, dispenser name needs to be Tap-A (as a default) in order to execute functions for Tap-A, any other dispenser name will default to Tap-B

**Review by PaoloLlenado — APPROVED** [review]

**Pull request merged**